src/api/system.c

```diff
@@ -851,19 +851,18 @@ static int f_get_fs_type(lua_State *L) {
 
 
 static int f_ftruncate(lua_State *L) {
-#if LUA_VERSION_NUM < 503
-  // note: it is possible to support pre 5.3 and JIT
-  //       since file handles are just FILE*  wrapped in a userdata;
-  //       but it is not standardized. YMMV.
-  #error luaL_Stream is not supported in this version of Lua.
-#endif
+#ifdef LUA_FILEHANDLE
```

Comment:
I wonder if you could do an early exit with `#ifndef`, without an `#else` block.
Would that raise unreachable code warnings?